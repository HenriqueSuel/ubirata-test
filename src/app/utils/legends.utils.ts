import { PopulationNeighborhoods } from "../interfaces/neighborhoodGeometries.interface";

import colors from "tailwindcss/colors";

const LEGENDS = [
  {
    value: 20000,
    text: "20000 - 25000",
    color: "before:bg-blue-950",
    rgb: colors.blue[950],
  },
  {
    value: 17000,
    text: "17000 - 19999",
    color: "before:bg-blue-800",
    rgb: colors.blue[800],
  },
  {
    value: 13000,
    text: "13000 - 16999",
    color: "before:bg-blue-500",
    rgb: colors.blue[500],
  },
  {
    value: 9000,
    text: "9000 - 12999",
    color: "before:bg-blue-300",
    rgb: colors.blue[300],
  },
  {
    value: 5000,
    text: "5000 - 8999",
    color: "before:bg-blue-100",
    rgb: colors.blue[100],
  },
];

const findColorByPopulation = (
  id: number,
  populations: PopulationNeighborhoods[],
  years: string
) => {
  const valueDefault = LEGENDS[LEGENDS.length - 1].rgb;
  
  const population = populations.find(
    (population) => population.id_geometria === id && years === population.ano
  );

  if (!population) return valueDefault;

  return LEGENDS.find((legend) => population?.populacao > legend.value)?.rgb;
};

export { LEGENDS, findColorByPopulation };
