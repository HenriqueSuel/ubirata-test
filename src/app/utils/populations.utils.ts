import {
  FeatureCollection,
  PopulationNeighborhoods,
} from "../interfaces/neighborhoodGeometries.interface";

export const handleSelectFilter = (
  neighborhoodGeometries: FeatureCollection,
  populationNeighborhoods: PopulationNeighborhoods[]
) => {
  const neighborhoods = filterNeighborhoods(neighborhoodGeometries);
  const years = filterUniqueYears(populationNeighborhoods);

  return { neighborhoods, years };
};

export const filterNeighborhoods = (neighborhoodGeometries: FeatureCollection) => {
  const neighborhoods = neighborhoodGeometries.features.map(
    (feature) => feature.properties
  );
  return neighborhoods;
};

export const filterUniqueYears = (
  populationNeighborhoods: PopulationNeighborhoods[]
) => {
  const uniqueYears = new Set(
    populationNeighborhoods.map((population) => population.ano)
  );
  const years = Array.from(uniqueYears);

  return years;
};
