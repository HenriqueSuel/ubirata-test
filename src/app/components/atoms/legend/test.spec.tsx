import { render, screen } from '@testing-library/react';
import { LEGENDS } from '../../../utils/legends.utils';
import { Legend } from './';

describe('Legend', () => {
  test('renders legend items correctly', () => {
    render(<Legend />);

    LEGENDS.forEach(({ value, text }) => {
      const legendItem = screen.getByText(text);
      expect(legendItem).toBeInTheDocument();
    });
  });
});
