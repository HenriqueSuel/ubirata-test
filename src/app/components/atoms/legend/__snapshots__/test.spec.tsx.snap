// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Legend > should renders Legend 1`] = `
<div
  class="max-w-sm rounded overflow-hidden shadow-lg absolute bottom-32 -right-1"
>
  <div
    class="px-1 py-4 bg-black"
  >
    <div
      class="before:content-[''] before:inline-block before:relative before:bg-blue-950 before:w-5 before:h-5 before:top-1 before:mr-3"
    >
      20000 - 25000
    </div>
    <div
      class="before:content-[''] before:inline-block before:relative before:bg-blue-800 before:w-5 before:h-5 before:top-1 before:mr-3"
    >
      17000 - 19999
    </div>
    <div
      class="before:content-[''] before:inline-block before:relative before:bg-blue-500 before:w-5 before:h-5 before:top-1 before:mr-3"
    >
      13000 - 16999
    </div>
    <div
      class="before:content-[''] before:inline-block before:relative before:bg-blue-300 before:w-5 before:h-5 before:top-1 before:mr-3"
    >
      9000 - 12999
    </div>
    <div
      class="before:content-[''] before:inline-block before:relative before:bg-blue-100 before:w-5 before:h-5 before:top-1 before:mr-3"
    >
      5000 - 8999
    </div>
  </div>
</div>
`;
